lib/fluent/plugin/filter_process_snmptrap.rb

```diff
@@ -38,27 +59,23 @@ def start
     end
 
     def filter(tag, time, record)
-      message = record.to_s
-      message = message.delete('\\"')
-      snmp_msg = message.gsub(/(?:(SNMPv2-(\w+)(::)(\w+)((\.)(\d+)){1,13}(=>))|(host=>))/, "")
```

Comment:
Can you specify the format of messages produced using this, and the format of messages after your change?

Reply:
When you convert a Hash to a string, it prepends '\' to each ". Since I stopped converting it to a string, that delete is not needed.
Second one was to remove some entries from the message. Not needed since the message now is a clone of the record. There is no need to delete anything.